src/resource-rental/resource-rental.ts

```diff
@@ -93,21 +107,35 @@ export class ResourceRental {
 
   /**
    * Creates an activity on the Provider, and returns a exe-unit that can be used to operate within the activity
+   * @param signalOrTimeout - timeout in milliseconds or an AbortSignal that will be used to cancel the exe-unit request,
+   * especially when the exe-unit is in the process of starting, deploying and preparing the environment (including setup function)
    */
-  async getExeUnit(): Promise<ExeUnit> {
+  async getExeUnit(signalOrTimeout?: number | AbortSignal): Promise<ExeUnit> {
     if (this.finalizePromise || this.abortController.signal.aborted) {
       throw new GolemUserError("The resource rental is not active. It may have been aborted or finalized");
     }
     if (this.currentExeUnit) {
       return this.currentExeUnit;
     }
 
+    const abortController = new AbortController();
+    this.abortController.signal.addEventListener("abort", () =>
+      abortController.abort(this.abortController.signal.reason),
+    );
+    if (signalOrTimeout) {
+      const abortSignal = createAbortSignalFromTimeout(signalOrTimeout);
+      abortSignal.addEventListener("abort", () => abortController.abort(abortSignal.reason));
+      if (signalOrTimeout instanceof AbortSignal && signalOrTimeout.aborted) {
+        abortController.abort(signalOrTimeout.reason);
+      }
+    }
```

Comment:
why do we need a new AbortController instance here? Can't we just pass `abortSignal` to `createExeUnit`?

Reply:
Because `getExeUnit()` should be aborted in two cases, via `signalOrTimeout` from the user parameter or via `abortController` from `ResourceRental` (e.g. if someone calls `stopAndFinalize()`.
That's why we need to listen to these two sources.

Reply:
Ok that makes sense.
Shame that `AbortSignal.any([signal1, signal2, ...])` is only from node 20 and up, it would be nice here

Reply:
Yes `AbortSignal.any` would be cool in this situation